octobot_trading/data_manager/prices_manager.py

```diff
@@ -28,34 +30,92 @@ def __init__(self, exchange_manager):
         self.logger = get_logger(self.__class__.__name__)
         self.mark_price = 0
         self.mark_price_set_time = 0
+        self.mark_price_from_sources = {}
         self.exchange_manager = exchange_manager
 
         # warning: should only be created in the async loop thread
         self.valid_price_received_event = Event()
 
     async def initialize_impl(self):
+        """
+        Initialize PricesManager attributes with default
+        """
         self._reset_prices()
 
-    def set_mark_price(self, mark_price):
-        self.mark_price = mark_price
-        self.mark_price_set_time = self.exchange_manager.exchange.get_exchange_current_time()
-        self.valid_price_received_event.set()
+    def set_mark_price(self, mark_price, mark_price_source):
+        """
+        Set the mark price if the mark price come from MarkPriceSources.EXCHANGE_MARK_PRICE
+        else add to mark_price_from_sources and perform calculation
+        :param mark_price: the mark price value from mark_price_source
+        :param mark_price_source: the mark_price source (from MarkPriceSources)
+        """
+        if mark_price_source == MarkPriceSources.EXCHANGE_MARK_PRICE.value:
+            self._set_mark_price_value(mark_price)
+        elif mark_price_source == MarkPriceSources.RECENT_TRADE_AVERAGE.value and \
+                self.mark_price_from_sources.get(MarkPriceSources.RECENT_TRADE_AVERAGE.value, None) is not None:
```

Comment:
I think we should add a comment here to explicit the fact that this allows to drop the 1st recent trades wake up

Reply:
On the other hand, maybe we should do this check directly on the price channel since it's because of it that we have to drop the 1st recent trades, what do you think ?

Reply:
We can but I think it's not the proper place for it. It's a feature related to mark price and not its updating process